testpar/t_filters_parallel.c

```diff
@@ -576,11 +576,22 @@ create_datasets(hid_t parent_obj_id, const char *dset_name, hid_t type_id, hid_t
         case USE_MULTIPLE_DATASETS:
         case USE_MULTIPLE_DATASETS_MIXED_FILTERED:
             dset_name_ptr = dset_name_multi_buf;
-            n_dsets       = (rand() % (MAX_NUM_DSETS_MULTI - 1)) + 2;
+
+            if (MAINPROCESS)
+                n_dsets = (rand() % (MAX_NUM_DSETS_MULTI - 1)) + 2;
```

Comment:
On some systems, other software in the stack appears to be causing rand() calls to get out of sync between ranks, which is responsible for hangs on those systems during the multi-dataset I/O tests. Broadcast these values from rank to ensure they're consistent across every rank